Subject: [PATCH] systemservice will be restarted if force stopping android.print.cts running

There are chances that handleDestroyPrinterDiscoverySession been
scheduled after binderDied() -- > handleBinderDied(), which will make
handleBinderDied() been called twice.

The exception log as below:
*** FATAL EXCEPTION IN SYSTEM PROCESS: main
Attempt to invoke interface method 'android.os.IBinder
android.printservice.IPrintService.asBinder()' on a null object reference
at com.android.server.print.RemotePrintService.handleBinderDied(RemotePrintService.java:163)
at com.android.server.print.RemotePrintService.-wrap1(RemotePrintService.java)
at com.android.server.print.RemotePrintService$MyHandler.handleMessage(RemotePrintService.java:705)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:154)
at com.android.server.SystemServer.run(SystemServer.java:349)
at com.android.server.SystemServer.main(SystemServer.java:219)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)

diff --git a/services/print/java/com/android/server/print/RemotePrintService.java b/services/print/java/com/android/server/print/RemotePrintService.java
index 9c3a852..2f9e72b 100644
--- a/services/print/java/com/android/server/print/RemotePrintService.java
+++ b/services/print/java/com/android/server/print/RemotePrintService.java
@@ -160,7 +160,8 @@
     }
 
     private void handleBinderDied() {
-        mPrintService.asBinder().unlinkToDeath(this, 0);
+        if (mPrintService != null)
+            mPrintService.asBinder().unlinkToDeath(this, 0);
         mPrintService = null;
         mServiceDied = true;
         mCallbacks.onServiceDied(this);
